
project device/sony/common/
diff --git a/hardware/healthd/healthd_board_sony.cpp b/hardware/healthd/healthd_board_sony.cpp
index 7f53193..d3e4363 100644
--- a/hardware/healthd/healthd_board_sony.cpp
+++ b/hardware/healthd/healthd_board_sony.cpp
@@ -52,7 +52,7 @@ struct led_ctl {
     const char *path;
 };
 
-struct led_ctl leds[3] =
+struct led_ctl leds_sony[3] =
     {{RED_LED, RED_LED_PATH},
     {GREEN_LED, GREEN_LED_PATH},
     {BLUE_LED, BLUE_LED_PATH}};
@@ -62,7 +62,7 @@ struct soc_led_color_mapping {
     int color;
 };
 
-struct soc_led_color_mapping soc_leds[3] = {
+struct soc_led_color_mapping soc_leds_sony[3] = {
     {15, RED_LED},
     {90, RED_LED | GREEN_LED},
     {100, GREEN_LED},
@@ -73,9 +73,9 @@ static int set_tricolor_led(int on, int color)
     int fd, i;
     char buffer[10];
 
-    for (i = 0; i < (int)ARRAY_SIZE(leds); i++) {
-        if ((color & leds[i].color) && (access(leds[i].path, R_OK | W_OK) == 0)) {
-            fd = open(leds[i].path, O_RDWR);
+    for (i = 0; i < (int)ARRAY_SIZE(leds_sony); i++) {
+        if ((color & leds_sony[i].color) && (access(leds_sony[i].path, R_OK | W_OK) == 0)) {
+            fd = open(leds_sony[i].path, O_RDWR);
             if (fd < 0) {
                 LOGE("Could not open led node %d\n", i);
                 continue;
@@ -94,21 +94,21 @@ static int set_tricolor_led(int on, int color)
     return 0;
 }
 
-static int set_battery_soc_leds(int soc)
+static int set_battery_soc_leds_sony(int soc)
 {
     int i, color;
     static int old_color = 0;
 
-    for (i = 0; i < (int)ARRAY_SIZE(soc_leds); i++) {
-        if (soc <= soc_leds[i].soc)
+    for (i = 0; i < (int)ARRAY_SIZE(soc_leds_sony); i++) {
+        if (soc <= soc_leds_sony[i].soc)
             break;
     }
-    color = soc_leds[i].color;
+    color = soc_leds_sony[i].color;
     if (old_color != color) {
         set_tricolor_led(0, old_color);
         set_tricolor_led(1, color);
         old_color = color;
-        LOGV("soc = %d, set led color 0x%x\n", soc, soc_leds[i].color);
+        LOGV("soc = %d, set led color 0x%x\n", soc, soc_leds_sony[i].color);
     }
     return 0;
 }
@@ -143,7 +143,7 @@ void healthd_board_mode_charger_battery_update(
 
     if (old_soc != soc) {
         old_soc = soc;
-        set_battery_soc_leds(soc);
+        set_battery_soc_leds_sony(soc);
     }
 }
 

project device/sony/common-kernel/
diff --git a/kernel-dtb-kugo b/kernel-dtb-kugo
index f61d4a6..f950119 100644
Binary files a/kernel-dtb-kugo and b/kernel-dtb-kugo differ

project hardware/broadcom/libbt/
diff --git a/src/bt_vendor_brcm.c b/src/bt_vendor_brcm.c
index 0251a89..3b569bd 100755
--- a/src/bt_vendor_brcm.c
+++ b/src/bt_vendor_brcm.c
@@ -32,6 +32,13 @@
 #include "upio.h"
 #include "userial_vendor.h"
 
+#include <cutils/properties.h>
+#include <fcntl.h>
+#include <termios.h>
+#include <sys/socket.h>
+#include <cutils/sockets.h>
+#include <linux/un.h>
+
 #ifndef BTVND_DBG
 #define BTVND_DBG FALSE
 #endif
@@ -259,10 +266,18 @@ static void ssr_cleanup( int reason )
 }
 
 // Entry point of DLib
+//const bt_vendor_interface_t BLUETOOTH_VENDOR_LIB_INTERFACE = {
+//    .size = sizeof(bt_vendor_interface_t),
+//    .init = init,
+//    .op = op,
+//    .cleanup = cleanup,
+//    .ssr_cleanup = ssr_cleanup,
+//};
 const bt_vendor_interface_t BLUETOOTH_VENDOR_LIB_INTERFACE = {
-    .size = sizeof(bt_vendor_interface_t),
-    .init = init,
-    .op = op,
-    .cleanup = cleanup,
-    .ssr_cleanup = ssr_cleanup,
+    sizeof(bt_vendor_interface_t),
+    init,
+    op,
+    cleanup,
+    ssr_cleanup
 };
+

project hardware/qcom/camera/
diff --git a/QCamera2/HAL/QCameraMem.cpp b/QCamera2/HAL/QCameraMem.cpp
index d4cf3ce..cd531ea 100755
--- a/QCamera2/HAL/QCameraMem.cpp
+++ b/QCamera2/HAL/QCameraMem.cpp
@@ -1850,7 +1850,8 @@ bool QCameraVideoMemory::needPerfEvent(const void *opaque, bool metadata)
             nh = (native_handle_t *)packet->pHandle;
         }
         isPerf = (MetaBufferUtil::getIntAt(nh, 0, VIDEO_META_EVENT) ==
-                CAM_META_BUFFER_EVENT_PERF) ? TRUE : FALSE;
+                //CAM_META_BUFFER_EVENT_PERF) ? TRUE : FALSE;
+                0x01 ) ? TRUE : FALSE;
 #else
 		if ((packet != NULL) && (packet->buffer_type ==
                 kMetadataBufferTypeNativeHandleSource)
@@ -1860,7 +1861,8 @@ bool QCameraVideoMemory::needPerfEvent(const void *opaque, bool metadata)
         for (int i = 0; i < mMetaBufCount; i++) {
             if(mMetadata[i]->data == opaque) {
                 isPerf = (MetaBufferUtil::getIntAt(nh, 0, VIDEO_META_EVENT) ==
-                        CAM_META_BUFFER_EVENT_PERF) ? TRUE : FALSE;
+                        //CAM_META_BUFFER_EVENT_PERF) ? TRUE : FALSE;
+                        0x01 ) ? TRUE : FALSE;
                 break;
             }
         }
diff --git a/QCamera2/HAL/QCameraMem.h b/QCamera2/HAL/QCameraMem.h
index 00f1c3e..9d74ad0 100644
--- a/QCamera2/HAL/QCameraMem.h
+++ b/QCamera2/HAL/QCameraMem.h
@@ -37,6 +37,7 @@
 
 //Media depedancies
 #include "OMX_QCOMExtns.h"
+#include "/home/kieran/halium/hardware/qcom/media-caf/msm8998/mm-core/inc/OMX_QCOMExtns.h"
 
 // Display dependencies
 #include "qdMetaData.h"
@@ -74,7 +75,8 @@ enum QCameraVideoMetaBufInts {
     VIDEO_META_TIMESTAMP   = MetaBufferUtil::INT_TIMESTAMP,
     VIDEO_META_FORMAT      = MetaBufferUtil::INT_COLORFORMAT,
     VIDEO_META_BUFIDX      = MetaBufferUtil::INT_BUFINDEX,
-    VIDEO_META_EVENT       = MetaBufferUtil::INT_BUFEVENT,
+    //VIDEO_META_EVENT       = MetaBufferUtil::INT_BUFEVENT,
+    VIDEO_META_EVENT       = 7, // work around a build issue
     //Add per frame Ints before this
     VIDEO_METADATA_NUM_INTS = MetaBufferUtil::INT_TOTAL,
 };

project kernel/sony/msm/
diff --git a/drivers/staging/wlan-qc/qcacld-3.0 b/drivers/staging/wlan-qc/qcacld-3.0
index 6c0c1f4c6..e3c64a38a 160000
--- a/drivers/staging/wlan-qc/qcacld-3.0
+++ b/drivers/staging/wlan-qc/qcacld-3.0
@@ -1 +1 @@
-Subproject commit 6c0c1f4c6fbfa5e72c543028dc659ddd433bacbf
+Subproject commit e3c64a38a47699c2004ee9cb7bed503931b7e0fc

project system/sepolicy/
diff --git a/untrusted_app.te b/untrusted_app.te
index 07bfbf4..4a46a0a 100644
--- a/untrusted_app.te
+++ b/untrusted_app.te
@@ -191,9 +191,9 @@ neverallow untrusted_app {
 
 # For exfat, f2fs/ext4, and ntfs storage via sdcardfs
 ifelse(shipping_build, `true', ,`
-  -exfat
+  #-exfat
   -fuseblk
-  -ntfs
+  #-ntfs
   -sdcard_posix
 ')
 
